xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/service/BTCMarketsStreamingMarketDataService.java

```diff
@@ -0,0 +1,52 @@
+package info.bitrich.xchangestream.btcmarkets.service;
+
+import static info.bitrich.xchangestream.btcmarkets.service.BTCMarketsStreamingService.CHANNEL_ORDERBOOK;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+import info.bitrich.xchangestream.btcmarkets.BTCMarketsStreamingAdapters;
+import info.bitrich.xchangestream.btcmarkets.dto.BTCMarketsWebSocketOrderbookMessage;
+import info.bitrich.xchangestream.core.StreamingMarketDataService;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import io.reactivex.Observable;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.marketdata.Ticker;
+import org.knowm.xchange.dto.marketdata.Trade;
+import org.knowm.xchange.exceptions.NotAvailableFromExchangeException;
+
+public class BTCMarketsStreamingMarketDataService implements StreamingMarketDataService {
+
+  private final ObjectMapper mapper = StreamingObjectMapperHelper.getObjectMapper();
+
+  private final BTCMarketsStreamingService service;
+
+  public BTCMarketsStreamingMarketDataService(BTCMarketsStreamingService service) {
+    this.service = service;
+  }
+
+  private OrderBook handleOrderbookMessage(BTCMarketsWebSocketOrderbookMessage message)
+      throws InvalidFormatException {
+    return BTCMarketsStreamingAdapters.adaptOrderbookMessageToOrderbook(message);
+  }
+
+  @Override
+  public Observable<OrderBook> getOrderBook(CurrencyPair currencyPair, Object... args) {
+    String marketId = BTCMarketsStreamingAdapters.adaptCurrencyPairToMarketId(currencyPair);
+    return service
+        .subscribeChannel(CHANNEL_ORDERBOOK, marketId)
+        .map(node -> mapper.treeToValue(node, BTCMarketsWebSocketOrderbookMessage.class))
+        .filter(orderEvent -> orderEvent.marketId != null && orderEvent.marketId.equals(marketId))
```

Comment:
could simplify this to `marketId.equals(orderEvent.marketId)` and lose the null check

Reply:
Done